src/redux/slices/productsCartSlice.js

```diff
@@ -0,0 +1,59 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+//Slice
+export const productsCartSlice = createSlice({
+  name: 'productsCartList',
+  initialState: {
+    productsCart: [],
+    cartQuantity: 0,
+  },
+  reducers: {
+    addProduct: (state, action) => {
+      for (let i = 0; i < state.productsCart.length; i++) {
+        if (state.productsCart[i].product.id === action.payload.product.id) {
+          state.productsCart[i].quantity += 1
+          state.cartQuantity += 1
+          return
+        }
+      }
+      const newProduct = {
+        quantity: action.payload.quantity,
+        product: action.payload.product,
+      }
+      state.cartQuantity += 1
+      state.productsCart.push(newProduct)
+    },
+    removeProduct: (state, action) => {
+      state.productsCart = state.productsCart.filter(
+        (item) => item.product.id !== action.payload.product.id
+      )
+      state.cartQuantity = state.cartQuantity - action.payload.quantity
+    },
+    changeProductQuantity: (state, action) => {
+      for (let i = 0; i < state.productsCart.length; i++) {
```

Comment:
creo que esta logica podrias ser mas funcional usando .map o reducer

Reply:
Creo que en este caso no serviria el .map o reducer porque no estoy modificando cada uno sino solo encontrando uno de la lista, modificandolo y me salgo del for.